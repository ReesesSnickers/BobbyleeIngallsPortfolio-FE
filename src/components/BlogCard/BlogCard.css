.portfolio-blogcard-image {
  object-fit: fill;
  width: 100%;
  height: 100%;
}
.portfolio-blogcard-image-wrapper {
  min-width: 180px;
  min-height: 140px;
  max-width: 180px;
  max-height: 140px;
  width: 180px;
  height: 140px;
  margin-left: 20px;
}
.portfolio-blogcard-wrapper {
  display: flex;
  flex-direction: row;
  width: 1150px;
  height: 170px;
  border-radius: 25px;
  background-color: #ffffff;
  border-color: #0e1a3c;
  padding: 0px;
  cursor: pointer;
}
.portfolio-blogcard-wrapper:focus {
  outline: none;
  border: 2px solid #eb21d4;
  box-shadow: 0px 0px 5px #eb21d4;
}
.portfolio-blogcard-title {
  color: #0e1a3c;
  text-align: left;
  max-width: 905px;
  font-size: 30px;
  font-family: Helvetica, Arial, sans-serif;
  margin: 0px;
}
.portfolio-blogcard-description {
  color: #0e1a3c;
  text-align: left;
  max-width: 905px;
  height: 70px;
  font-size: 20px;
  font-family: Helvetica, Arial, sans-serif;
  margin: 0px;
  -webkit-box-orient: vertical;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: normal;
}
.portfolio-blogcard-created {
  color: #0e1a3c;
  text-align: right;
  font-size: 20px;
  font-family: Helvetica, Arial, sans-serif;
  margin: 15px 0px 0px 0px;
}
.portfolio-blogcard-details-wrapper {
  margin-left: 20px;
  width: 905px;
  height: 140px;
}
