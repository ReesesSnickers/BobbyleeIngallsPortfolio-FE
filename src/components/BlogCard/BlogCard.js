import React from "react";
import tempPhoto from "../../static/ProfilePhoto.jpg";

const BlogCard = ({ blog, onClick }) => {
  return (
    <button className="portfolio-blogcard-wrapper" onClick={onClick}>
      <div className="portfolio-blogcard-image-wrapper">
        <img
          className="portfolio-blogcard-image"
          // src={blog.photo}
          src={tempPhoto}
          alt={blog.title}
        />
      </div>
      <div className="portfolio-blogcard-details-wrapper">
        <h1 className="portfolio-blogcard-title">{blog.title}</h1>
        <h2 className="portfolio-blogcard-description">{blog.description}</h2>
        <h3 className="portfolio-blogcard-created">{blog.created}</h3>
      </div>
    </button>
  );
};

export default BlogCard;
