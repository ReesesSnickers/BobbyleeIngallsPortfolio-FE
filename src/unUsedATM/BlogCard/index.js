import "./BlogCard.css";
export { default } from "./BlogCard";
