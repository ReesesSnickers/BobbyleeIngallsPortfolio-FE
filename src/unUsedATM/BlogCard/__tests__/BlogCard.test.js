import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import BlogCard from "../BlogCard.js";

let props;
let mountedComponent;

describe("BlogCard", () => {
  const testComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<BlogCard {...props} />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    props = {
      blog: undefined,
      onClick: undefined
    };
    mountedComponent = undefined;
  });

  // initial render
  it("Renders the BlogCard component", () => {
    let button = testComponent().find("button.portfolio-blogcard-wrapper");
    expect(button.length).toBe(1);
  });

  // snapshot
  it("should match snapshot", () => {
    const tree = renderer.create(<BlogCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // blog default
  it("Will use the default blog object if no blog is defined", () => {
    const image = testComponent().find("img");
    const h1 = testComponent().find("h1");
    const h2 = testComponent().find("h2");
    const h3 = testComponent().find("h3");
    expect(image.props().alt).toBe("");
    // expect(image.props().src).toBe('')
    expect(h1.text()).toBe("");
    expect(h2.text()).toBe("");
    expect(h3.text()).toBe("");
  });

  // blog defined
  describe("When a blog is defined", () => {
    beforeEach(() => {
      props.blog = {
        title: "MockTitle",
        description: "MockDescription",
        created: "MockCreated",
        photo: "MockPhoto"
      };
    });
    it("Will use the defined blog object", () => {
      const image = testComponent().find("img");
      const h1 = testComponent().find("h1");
      const h2 = testComponent().find("h2");
      const h3 = testComponent().find("h3");
      expect(image.props().alt).toBe("MockTitle");
      // expect(image.props().src).toBe('MockPhoto')
      expect(h1.text()).toBe("MockTitle");
      expect(h2.text()).toBe("MockDescription");
      expect(h3.text()).toBe("MockCreated");
    });
  });

  // onClick
  describe("When onClick is defined", () => {
    beforeEach(() => {
      props.onClick = jest.fn();
    });
    it("will trigger the onclick event when clicked", () => {
      const button = testComponent().find("button");
      button.simulate("click");
      expect(button.props().onClick).toHaveBeenCalled();
    });
  });
});
