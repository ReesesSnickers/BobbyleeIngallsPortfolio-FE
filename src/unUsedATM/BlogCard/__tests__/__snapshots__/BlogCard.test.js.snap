// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard should match snapshot 1`] = `
<button
  className="portfolio-blogcard-wrapper"
>
  <div
    className="portfolio-blogcard-image-wrapper"
  >
    <img
      alt=""
      className="portfolio-blogcard-image"
      src="ProfilePhoto.jpg"
    />
  </div>
  <div
    className="portfolio-blogcard-details-wrapper"
  >
    <h1
      className="portfolio-blogcard-title"
    >
      
    </h1>
    <h2
      className="portfolio-blogcard-description"
    >
      
    </h2>
    <h3
      className="portfolio-blogcard-created"
    >
      
    </h3>
  </div>
</button>
`;
